### ISSUE TITLE
Inconsistent highlighting of DataTypes

### ISSUE THREAD

**Issue by garborg** [JuliaLang/atom-language-julia#21]
Highlighting every other of the following is worse than no highlighting at all:
```julia
x = [Int, Bool, Vector, Array, Integer, FloatingPoint, DataArray, DataFrame]
```

Seems like consistent options are:

+ Highlight every type exported from Base
+ Go farther highlighting everything that's proven to be a type by syntax (e.g. `x = Union{...............}`)
+ Both of the above
+ Remove `'storage.type.julia'`

I think the latter is the cleanest and means the least distracting false alarms, but I wouldn't be that surprised if others like the second option.

Thoughts?

**Comment by spencerlyon2** [comment]
If the second option is doable without false positives, I think people might like it.  

Otherwise I agree that 4 is the cleanest.

To try to get 2, we might just _impose_ the [official naming conventions](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#general-formatting-guidelines-for-julia-code-contributions) from the contributing guide and say something like `(A-Z\w+)` is a type name (the initial `A-Z` enforces the upper-camel-case guideline), but I suspect that some users wouldn't like that. Also, it would be hard to disambiguate module and type names if we tried that.

**Comment by garborg** [comment]
I think this is closed by #22.

I think you're right about disambiguating module and type names, and possibly constants (in the case of single letters or acronyms). Also, type highlighting gets really loud if all the types being passed around as values are highlighted, and highlighting the type in `x = Union{Int, UInt}` but not `x = Int` seems a little strange, for example.
